File: frontend/lib/features/flower/domain/entities/flower.dart
import 'package:equatable/equatable.dart';

/// Pure Dart business entity for Flower
class Flower extends Equatable {
  const Flower({
    required this.id,
    required this.name,
    this.latinName,
    this.imageUrl,
    this.description,
    this.season,
  });

  final int id;
  final String name;
  final String? latinName;
  final String? imageUrl;
  final String? description;
  final String? season;

  @override
  List<Object?> get props => [id, name, latinName, imageUrl, description, season];
}

File: backend/app/models/flower.py
from datetime import datetime
from sqlalchemy import String, DateTime, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.database.connection import Base


class Flower(Base):
    __tablename__ = "flowers"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    latin_name: Mapped[str] = mapped_column(String(150), nullable=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    image_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    season: Mapped[str | None] = mapped_column(String(50), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

    detection_results: Mapped[list["DetectionResult"]] = relationship(
        "DetectionResult",
        back_populates="flower",
    )
